File: engine/engine-model/src/main/scala/tools/packet/message/Whisper.scala
package tools.packet.message

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class Whisper(private var _sender: String, private var _channel: Int, private var _text: String) extends PacketInput {
  def sender: String = _sender

  def channel: Int = _channel

  def text: String = _text

  override def opcode(): SendOpcode = SendOpcode.WHISPER
}

File: engine/engine-model/src/main/java/tools/packet/message/Whisper.java
package tools.packet.message;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record Whisper(String sender, Integer channel, String text) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.WHISPER;
   }
}